builder/proxmox/common/config.go

```diff
@@ -229,6 +230,15 @@ func (c *Config) Prepare(upper interface{}, raws ...interface{}) ([]string, []st
 			errs = packersdk.MultiErrorAppend(errs, fmt.Errorf("disk format must be specified for pool type %q", c.Disks[idx].StoragePoolType))
 		}
 	}
+	if len(c.Serials) > 4 {
+	    errs = packersdk.MultiErrorAppend(errs, fmt.Errorf("serials admit up to 4 elements"))
+	}
+	res := regexp.MustCompile(`^(/dev/.+|socket)$`)
+	for idx := range c.Serials {
```

Comment:
If you have no use for the index in any place, you can probably change this loop to
```suggestion
	for _, serial := range c.Serials {
```

This way `serial` will be each entry in the `c.Serials` array, and you don't have to access it through the index in the following block.